### PR TITLE
Add ref properly to input components

### PR DESCRIPTION
**Pull request by mickaelzhang**
## Screenshot / video

[if applicable for UI changes]

## What does this do?

[A short description of what the PR changes and why it's necessary]

- Give the reviewers context
- Is the impact global or localised?
- Any tech debt created or discovered?
- Include links to Jira ticket and Figma file if applicable

### CODE REVIEW

File: src/Textarea/Textarea.tsx [hunk 1]
```diff
@@ -12,8 +12,6 @@ type BaseProps = {
   id: string
   /** className attribute to apply classses from props */
   className?: string
-  /** ref attribute for input */
-  ref?: RefObject<HTMLTextAreaElement>
```
Comment (mickaelzhang) [inline src/Textarea/Textarea.tsx:16]: Adding it to the props shouldn't do anything, I've wrapper the component in `forwardRef` instead

File: src/SearchInput/SearchInput.tsx [hunk 2]
```diff
@@ -195,6 +195,8 @@ const Input = styled.input<ISearchInput>`
 
   &::placeholder {
     color: ${theme.colors.subtext};
+  }
+
```
Comment (mickaelzhang) [inline src/SearchInput/SearchInput.tsx:199]: It seems to have been missing since the color migration

**Comment by fleonard** [comment]
@mickaelzhang I'm wondering if on top of forwarding ref we should also forward all the other missing props in order to be able to use data attributes and such? I've already done it in the Button and Chip components.
Just not 100% sure what exactly we should fw since the form elements are nested

**Comment by mickaelzhang** [comment]
@fleonard I think we should but it's definitely trickier as you said, there will probably be the need to split the props around either the container component and the input 🤔 

I don't think it's related to this PR so maybe we can take a look at it separately?

**Comment by fleonard** [comment]
> @fleonard I think we should but it's definitely trickier as you said, there will probably be the need to split the props around either the container component and the input 🤔
> 
> I don't think it's related to this PR so maybe we can take a look at it separately?

@mickaelzhang yeah definitely not part of this PR. Was just a random question if you had some sort of idea already

**Review by fleonard — APPROVED** [review]

**Pull request merged**